sysdeps/managarm/generic/file.cpp

```diff
@@ -4088,5 +4089,20 @@ int sys_utimensat(int dirfd, const char *pathname, const struct timespec times[2
 	}
 }
 
+int sys_getentropy(void *buffer, size_t length) {
+	SignalGuard sguard;
+	uint8_t buf[length];
+	size_t size;
+
+	if(length > 256)
+		return EIO;
+
+	HEL_CHECK(helGetRandomBytes(buf, length, &size));
+
+    memcpy(buffer, buf, size);
```

Comment:
Here we have an issue since `helGetRandomBytes` might return fewer than `length` bytes, while `getentropy` is supposed to return exactly `length` bytes. We also do not need `buf`. Suggestion:

```suggestion
	auto p = reinterpret_cast<char *>(buffer);
	size_t n = 0;
	while(n < length) {
		size_t chunk;
		HEL_CHECK(helGetRandomBytes(p + n, length - n, &chunk));
		n += chunk;
	}
```